### PR TITLE
virtio_transitional: Add test case for virtio nic device

### PR DESCRIPTION
**Pull request by vivianQizhu**
Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+    def check_plug_to_pci_bridge(vm_name, mac):
+        """
+        Check if the nic is plugged onto pcie-to-pci-bridge
+
+        :param vm_name： Vm name
+        :param mac:  The mac address of plugged interface
+        :return True if plugged onto pcie-to-pci-bridge, otherwise False
+        """
+        v_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
```
Comment (vivianQizhu) [inline libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: @kylazhang Hi, I am not able to get the plugged interface this way, could you help indicate how to get it from xml?

Comment (kylazhang) [reply libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: new_from_dumpxml(vm_name), remove "inactive"

Comment (vivianQizhu) [reply libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: @kylazhang Okay, thanks a lot.

**Comment by vivianQizhu** [comment]
@kylazhang Updated, please help review, thanks.
 (1/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.default.virtio.boot_test: PASS (104.18 s)
 (2/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.default.virtio_non_transitional.boot_test: |PASS (166.74 s)
 (3/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.boot_test: PASS (63.13 s)
 (4/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.attach_device_test: PASS (84.29 s)
 (5/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.attach_interface_test: -\PASS (78.02 s)
 (6/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.coldplug_test: PASS (153.68 s)
 (7/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.save_restore: PASS (77.76 s)

File: libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            if process.system(download_cmd, shell=True):
+                test.cancel("Failed to download file")
+
+        image_name = params['image_path']
+        target_path = utils_misc.get_path(data_dir.get_data_dir(), image_name)
+        if not os.path.exists(target_path):
+            if utils_package.package_install("wget"):
+                utils_misc.wait_for(_download, timeout=360)
+            else:
+                test.error("Fail to install wget")
```
Comment (kylazhang) [inline libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: May be no needed if use avocado lib

File: libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    pci_controllers = vmxml.get_controllers('pci')
+    for controller in pci_controllers:
+        if controller.get('model') == 'pcie-to-pci-bridge':
+            break
+    else:
+        contr_dict = {'controller_type': 'pci',
+                      'controller_model': 'pcie-to-pci-bridge'}
+        libvirt.create_controller_xml(contr_dict, "add_controller", vm_name)
+    try:
+        iface_params = {'type': 'bridge', 'model': params['virtio_model']}
```
Comment (kylazhang) [inline libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: please remove type:bridge if not forced. Else need to set more for source

File: libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        except_mesg = ''
+        try:
+            if guest_os_type == 'Windows':
+                utils_net.restart_windows_guest_network_by_key(session,
+                                                               "macaddress",
+                                                               mac)
+                except_mesg = "Get nic netconnectionid failed"
+            else:
+                utils_net.get_linux_ifname(session, mac)
+                except_mesg = ("Failed to determine interface"
```
[the remark this answers is not in the document]
Comment (vivianQizhu) [reply libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: @kylazhang No it does not need, `except_mesg` only use for L#152 below, the sequence does not matter.

[the remark this answers is not in the document]
Comment (vivianQizhu) [reply libvirt/tests/src/virtio_transitional/virtio_transitional_nic.py]: @kylazhang Oh, I get your point, yes it should, will update. Thanks.

**Comment by vivianQizhu** [comment]
@kylazhang Updated,
1. Utilize `utils_net` for renew ip address;
2. Remove `{'type': 'bridge'}`.
3. Utilize `download.get_file()`
4. Move `except_mesg` definition before where the exception might been raised.
5. Add some comments to explain steps.

Please help review, thanks.
```
 (1/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.default.virtio.boot_test: PASS (125.17 s)
 (2/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.default.virtio_non_transitional.boot_test: PASS (101.79 s)
 (3/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.boot_test: PASS (63.65 s)
 (4/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.attach_device_test: PASS (83.59 s)
 (5/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.attach_interface_test: PASS (77.64 s)
 (6/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.hotplug.coldplug_test: PASS (152.69 s)
 (7/7) type_specific.io-github-autotest-libvirt.virtio_transitional_nic.rhel6_guest.virtio_transitional.save_restore: PASS (74.74 s)
```

**Pull request merged**